ambry-store/src/main/java/com/github/ambry/store/LogSegment.java

```diff
@@ -214,6 +214,29 @@ private DirectIOExecutor createDirectIOExecutor(int bufferSize) {
     return new DirectIOBufferedExecutor(bufferSize);
   }
 
+  /**
+   * Truncate the log segment file to the given offset. If the endOffset is already set, then this operation
+   * should be illegal.
+   * @param offset
+   * @throws IOException
+   */
+  void truncateTo(long offset) throws IOException {
+    long currentFileSize = sizeInBytes();
+    if (currentFileSize < offset) {
```

Comment:
QQ: in which scenario this will happen?

Reply:
it should never happen, adding it here just for sanity check.